components/admin_console/permission_schemes_settings/edit_post_time_limit_button/edit_post_time_limit_button.tsx

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {Constants} from 'utils/constants';
+import {t} from 'utils/i18n';
+
+type Props = {
+    timeLimit: number;
+    onClick: () => void;
+    isDisabled: boolean | undefined;
+}
+export default function EditPostTimeLimitButton(props: Props) {
+    let messageID;
+    if (props.timeLimit === Constants.UNSET_POST_EDIT_TIME_LIMIT) {
+        messageID = t('edit_post.time_limit_button.no_limit');
+    } else {
+        messageID = t('edit_post.time_limit_button.for_n_seconds');
+    }
```

Comment:
@jespino do we still need `t` function, are the ids still not able to be extracted with the mmjstool

Reply:
mmjstool can only extract IDs that are passed directly into one of the translation functions as constants. If there's any conditional logic like this, we need to use `t` to mark them

Reply:
Thank you @hmhealey for clarifying this.